shared/detect-event-type.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+#  Event Types:
+#   - BuildEvent
+#   - MergeEvent
+#   - VerisonEvent
+#   - RebaseEvent
+#
+# The event type is detected based on the action and type of the github-events-resource event
+# These represent the events and types of the events emitted by the github API which are then
+# mapped to the event types above
+
+set -e
+
+ARGV_DIRECTORY="$1"
+
+pushd "$ARGV_DIRECTORY" > /dev/null
+
+action="$(jq -r '.action' .git/.version)"
+type="$(jq -r '.type' .git/.version)"
+
+function emit_build_event() {
+  echo "BuildEvent"
+}
+
+function emit_merge_event() {
+  echo "MergeEvent"
+}
+
+function emit_version_event() {
+  echo "VersionEvent"
+}
+
+function emit_rebase_event() {
+  echo "RebaseEvent"
+}
+
+if [ "$type" == "PullRequestEvent" ]; then
```

Comment:
What about replacing the if/else with a case statement? :) (https://tldp.org/LDP/Bash-Beginners-Guide/html/sect_07_03.html)

Reply:
I don't really see the benefit in this case, as there is no smart pattern-matching to really do, do you have a strong argument for that?

Reply:
Yep, readability and therefore maintainability (there's no technical issue). As an suggestion from the top of my head:

```suggestion
 case "$type" in
    "PullRequestEvent")
        if [ "${action}" == "merged" ]; then
            emit_merge_event
        else
            emit_build_event
        fi
    ;;
    "IssueCommentEvent")
        if [ "$action" == "retest" ]; then
            emit_build_event
        elif [ "$action" == "rebase" ]; then
            emit_rebase_event
        fi
    ;;
    "PushEvent")
        emit_build_event
    ;;
    "VersionEvent")
        emit_version_event
    ;;
esac```

And this can lead you to further refactoring, like getting rid of the if/else or replacing the functions by simple echo's since now all is self-contained